test/functional/wallet_assumeutxo.py

```diff
@@ -192,6 +198,18 @@ def run_test(self):
         w = n1.get_wallet_rpc("w")
         assert_equal(w.getbalance(), 34)
 
+        # Check balance of wallet on node 2 (for which wallet w is active during snapshot completion)
+        n2.restorewallet("w", "backup_w.dat")
+        self.log.info(
+            f"Loading snapshot into third node from {dump_output['path']}")
+        loaded = n2.loadtxoutset(dump_output['path'])
+        self.log.info("Ensuring background validation completes")
+        self.connect_nodes(0, 2)
+        self.wait_until(lambda: len(n2.getchainstates()['chainstates']) == 1)
+
+        self.log.info("Check balance at third node")
+        ensure_for(duration=1, f=lambda: (n2.getbalance() == 34))
```

Comment:
I like to see `ensure_for` getting used but I think I would have opted for `self.wait_until(lambda: (n2.getbalance() == 34), timeout=5)` or so here. Is there a case I am missing where the balance might change after having been 34 that you are covering with this?

Reply:
I used `ensure_for` because otherwise the added test might not always have failed on master before (if the first check had been done immediately, before the incorrect notification was processed by the wallet, the balance would have still been 34).